core-rust/core-api-server/core-api-schema.yaml

```diff
@@ -821,6 +821,10 @@ components:
           $ref: "#/components/schemas/FeeSummary"
         state_updates:
           $ref: "#/components/schemas/StateUpdates"
+        events:
```

Comment:
I imagine all receipts will include this list? So we'd want to include `events` in the required list of this object on the schema?

Reply:
Ahh I see we use this for rejected responses in the preview. Hmmm.

I wonder if maybe we should change preview to use an enum/tagged union in its result, to return a different object upon rejection, so that this can just be for committed (success/failure) cases?

(Not necessarily in this PR though!). What are your thoughts on this?

Reply:
yes, exactly (I initially made it required, but then the current preview's code convinced me to do optional...)
This (new shape of the preview's result) is a material for a separate PR, since it will have to be synchronized with the NG (at least regarding deploy).

I think ideally I would see something like:
```
preview(tx) -> Response(Rejected(error_message) | Accepted(receipt));
get_receipt(tx) -> Response(receipt);
```
My doubts:
1. The `receipt` in both cases would be the same api type, right?
2. I assumed that the opposite of `Rejected` is `Accepted`, to avoid using the word `Committed` in preview.

Reply:
Yep, exactly. Maybe `Accepted` would be `WouldBeCommitted` or something, to save introducing new nomenclature that people need to understand :)